### PR TITLE
Adds a recipe detailing how to join tables in DB-Connected resources

### PR DESCRIPTION
**Pull request by weierophinney**
Per zfcampus/zf-apigility#198 and zfcampus/zf-apigility#64, this patch provides a tutorial demonstrating adding a LEFT JOIN to create and populate entities returned by a DB-Connected REST resource.

### CODE REVIEW

File: recipes/join-tables.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+$statement = $sql->prepareStatementForSqlObject($select);
+$results = $statement->execute();
+```
+
+If you are using a normal REST resource (_not_ DB-Connected), and using zend-db
+or a table gateway to back it, you can likely pull the above into your existing
+resource, and, with a little work, have it returning your entities and
+collections.
+
+For DB-Connected resources, though, we now need to do a little work to integrate
```
Comment (akrabat) [inline recipes/join-tables.md]: Don’t need “to do a little work”

File: recipes/join-tables.md [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+```
+
+We now have a table gateway capable of producing what we need. We now need to
+tell the resource to use these new methods.
+
+### Updating the resource
+
+As noted earlier, DB-Connected services have a "virtual" resource. zf-apigility
+creates a `ZF\Apigility\DbConnectedResource` instance that performs the various
+operations. We need to somehow alter this to use our new table gateway, and also
```
Comment (akrabat) [inline recipes/join-tables.md]: Lose “somehow”.

File: recipes/join-tables.md [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+use the new methods it presents. How do we do that?
```
Comment (akrabat) [inline recipes/join-tables.md]: I’m not a fan of this sentence as it doesn’t really describe what we want to happen.

We don’t want to alter the virtual resource, we want to replace it with our own resource class that adds our new additonal functionality to what we get with the DbConnectedResource.

No idea how to phrase that though, and the current sentence can be lived with.

File: recipes/join-tables.md [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+    public function fetchAll($data = [])
+    {
+        return new UsersCollection($this->table->getUsersWithCities());
+    }
+}
+```
+
+What is this class doing?
+
+In `fetch()`, it pulls a result set by executing the table gateway's
```
Comment (akrabat) [inline recipes/join-tables.md]: “pulls” seems like the wrong word.

**Comment by zrecore** [comment]
This documentation makes things easier to understand. Thank you.

**Pull request merged**